src/main/java/com/pnu/dev/shpaltaif/aop/ModelAttributeSetter.java

```diff
@@ -0,0 +1,34 @@
+package com.pnu.dev.shpaltaif.aop;
+
+import com.pnu.dev.shpaltaif.domain.Category;
+import com.pnu.dev.shpaltaif.service.CategoryService;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ModelAttribute;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+@ControllerAdvice
+public class ModelAttributeSetter {
+
+    private CategoryService categoryService;
+
+    @Autowired
+    public ModelAttributeSetter(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    @ModelAttribute
+    public void setCategories(HttpServletRequest request, Model model) {
+
+        if (!StringUtils.startsWith(request.getRequestURI(), "/admin")) {
+            List<Category> categories = categoryService.findAll();
+            model.addAttribute("categories", categories);
+        }
+
+    }
```

Comment:
Is it called for each request?

Reply:
Yes, it does.
The idea here is next: we need to render sidebar with categories on each page, except pages related to admin panel.
So this logic just updates the model with categories attribute for each request that is not related to admin panel.